test_runner/regress/test_download_extensions.py

```diff
@@ -114,21 +193,18 @@ def handler(request: Request) -> Response:
     env.create_branch("test_remote_extensions")
     endpoint = env.endpoints.create("test_remote_extensions")
 
-    with open(extension_dir / "test_extension.control", encoding="utf-8") as f:
-        control_data = f.read()
-
     # mock remote_extensions spec
     spec: dict[str, Any] = {
-        "public_extensions": ["test_extension"],
+        "public_extensions": [extension],
         "custom_extensions": None,
         "library_index": {
-            "test_extension": "test_extension",
+            extension: extension,
```

Comment:
just an thought:
to improve coverage even more we can use library name that is different from extension name.
though, codepath with matching names is definitely more common.

Reply:
Hmm. I'm not entirely sure how to trigger the code path you're referring to. I need `is_library` to be true, but I don't know how to make a test like that. I'll merge this PR and if you can provide some more information, I'll be happy to extend it.